CONTRIBUTING.md

```diff
@@ -31,3 +31,32 @@ In general, here's a suggested workflow:
 6. Actively participate in your pull request.
    If the maintainers ask for a revision to your proposed changes, make it promptly to help get your PR merged as soon as possible.
    Inactive PR's or Issues may be marked as Deferred or closed if no response is given.
+
+## Branching/Versioning Scheme
+
+***Note: This guide is intended mainly for core contributors.***
+
+Dot uses [SemVer v2](http://semver.org/spec/v2.0.0.html) as a model to standardize our versioning system.
+A "breaking change" in our context is something that might cause a user to have to change their personal config; any change to the "API" (which is the set of functions exposed to the user) is considered as such.
+
+In general, our branching workflow consists of two persistent branches:
+
+- `master`, which contains all of the latest changes and should be in a stable state.
+  This is used as the base for all of the `release-` branches (see below.)
+- `stable`, which is a tracking branch that gets releases merged into it.
+  This is the branch to be used by users who only want stable features to be added to their configuration.
+
+In addition, we use the following scheme for additional branches:
+
+- `release-vX.Y` and `release-vX.Y.Z` branches are branches used to control versioning.
+  - To release a new version from `master`,
+    1. after merging all PR's into `master`,
+    2. Create your `release-v` branch.
+    3. Bump the version on the `release-v` branch, either through a PR on a separate branch or something like that.
+       If you have any release notes to add, do so here.
+    4. Merge the branch back into `master` via a PR, including writing release notes and such.
```

Comment:
I would agree if you're saying that any development work off of `stable` should take place and be directed back to the `stable` branch, but some people like @cg505 and myself would probably be working off of `master`.  Again, making a `release-vX.Y` branch is intended for work _related to the release process_, (incl. merging PR's) and nothing else.  The point of branching off of one of these branches is to have a nice place to merge back into, but I don't expect 3rd parties (or even contributors who are submitting their own changes) to properly take care of documenting their changes.

I don't understand what you mean by
>The diff should be 0.